pkg/http/client_test.go

```diff
@@ -0,0 +1,74 @@
+package http
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/dcos/dcos-cli/pkg/config"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGet(t *testing.T) {
+	query := `{"cluster":"DC/OS"}`
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte(query))
+	}))
+	defer ts.Close()
+
+	conf, err := config.FromString(`
+[core]
+
+dcos_url = "` + ts.URL + `"
+`)
+	require.NoError(t, err)
+
+	client := New(conf)
+
+	resp, err := client.Get("/")
+	require.NoError(t, err)
+
+	body, err := ioutil.ReadAll(resp.Body)
+	require.Equal(t, query, string(body))
+}
+
+func TestPost(t *testing.T) {
+	type query_struct struct {
+		Cluster string
+	}
+	query := `{"cluster":"DC/OS"}`
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "POST" {
+			t.Errorf("Expected ‘POST’ request, got ‘%s’", r.Method)
+		}
```

Comment:
As we started using it and for consistency across our tests, let's use testify here. It turns these in a one-liner :slightly_smiling_face: